File: examples/weather.ts
import { Agent, ModelRegistry, OpenAICompatibleAdapter, Runner } from "../sdks/javascript/src/index.ts";

const agent = new Agent({
  id: "weather", name: "Weather", instructions: "Use the weather tool and answer briefly.",
  model: "openai:gpt-5-mini", tools: [{
    name: "weather", description: "Get temperature",
    inputSchema: { type: "object", properties: { city: { type: "string" } }, required: ["city"] },
    execute: (arguments_: any) => ({ city: arguments_.city, temperature_c: 31 }),
  }],
});
const result = await new Runner(new ModelRegistry([new OpenAICompatibleAdapter()])).run(agent, "Weather in Delhi?");
console.log(result.output);

File: examples/weather.py
import asyncio
from orion_sdk import Agent, ModelRegistry, OpenAICompatibleAdapter, Runner, Tool

async def main() -> None:
    weather = Tool("weather", "Get temperature", {
        "type": "object", "properties": {"city": {"type": "string"}}, "required": ["city"],
    }, lambda arguments: {"city": arguments["city"], "temperature_c": 31})
    agent = Agent("weather", "Weather", "Use the weather tool and answer briefly.",
                  "openai:gpt-5-mini", tools=(weather,))
    result = await Runner(models=ModelRegistry([OpenAICompatibleAdapter()])).run(agent, "Weather in Delhi?")
    print(result.output)

asyncio.run(main())
